src/couch/src/couch_httpd_auth.erl

```diff
@@ -99,6 +99,13 @@ basic_name_pw(Req) ->
             nil
     end.
 
+extract_roles(UserProps) ->
+    Roles = couch_util:get_value(<<"roles">>, UserProps, []),
+    case lists:member(<<"_admin">>, Roles) of
+        true -> Roles;
+        _ -> [<<"_users">> | Roles]
```

Comment:
@rnewson I switched to this from `Roles ++ <<"_users">>` by your request, but this prepends the new role to the list. I’d rather append it as not to break any existing scripts (I had to update a few tests to match for this change already). Would you be okay going back to `++` or is there a better option?

Reply:
hi. going back to ++ is fine if we're confident the list stays quite short, which seems highly likely.